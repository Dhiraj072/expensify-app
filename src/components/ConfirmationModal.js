import React from 'react';
import Modal from 'react-modal';

export default (props) => (
    <Modal
        isOpen={props.isModalOpen}
        contentLabel="Remove expense confirmation"
        closeTimeoutMS={300}
        className="modal"
    >
        <div>
            <p>Remove expense?</p>
            <button
                value="yes"
                className="button button--margin"
                onClick={props.handleConfirmation}
            >
                Yes
            </button>
            <button
                value="no"
                className="button button--margin"
                onClick={props.handleConfirmation}
            >
                No
            </button>
        </div>
    </Modal>
);
