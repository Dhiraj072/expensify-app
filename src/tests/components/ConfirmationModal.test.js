import React from 'react';
import { shallow } from 'enzyme';
import ConfirmationModal from '../../components/ConfirmationModal';

let isModalOpen;
let handleConfirmation;
let wrapper;

beforeEach(() => {
    isModalOpen = true;
    handleConfirmation = jest.fn();
    wrapper = shallow(<ConfirmationModal
        isModalOpen={isModalOpen}
        handleConfirmation={handleConfirmation}
    />);
});

test('should render Confirmation Modal correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle yes confirmation from user', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(handleConfirmation).toHaveBeenCalled();
});

test('should handle no confirmation from user', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(handleConfirmation).toHaveBeenCalled();
});
