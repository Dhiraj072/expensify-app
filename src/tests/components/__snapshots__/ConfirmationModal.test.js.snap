// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Confirmation Modal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={300}
  contentLabel="Remove expense confirmation"
  isOpen={true}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <div>
    <p>
      Remove expense?
    </p>
    <button
      className="button button--margin"
      onClick={[Function]}
      value="yes"
    >
      Yes
    </button>
    <button
      className="button button--margin"
      onClick={[Function]}
      value="no"
    >
      No
    </button>
  </div>
</Modal>
`;
